sdk/storage/Azure.Storage.Common/src/UriExtensions.cs

```diff
@@ -59,5 +59,54 @@ public static Uri AppendToPath(this Uri uri, string segment)
             }
             return parameters;
         }
+
+        /// <summary>
+        /// Get the account name from the domain portion of a Uri.
+        /// </summary>
+        /// <param name="uri">The Uri.</param>
+        /// <param name="serviceSubDomains">Required. The service subdomains used to validate that the
+        /// domain is in the expected format. This should be "blob" for blobs, "file" for files,
+        /// "queue" for queues, "blob" and "dfs" for datalake.</param>
+        /// <returns>Account name</returns>
+        public static string GetAccountNameFromDomain(this Uri uri, params string[] serviceSubDomains)
+        {
+            var accountEndIndex = uri.Host.IndexOf(".", StringComparison.InvariantCulture);
+            if (accountEndIndex >= 0)
+            {
+                var serviceStartIndex = accountEndIndex + 1;
+                var serviceEndIndex = uri.Host.IndexOf(".", serviceStartIndex, StringComparison.InvariantCulture);
+                if (serviceEndIndex > serviceStartIndex)
+                {
+                    var service = uri.Host.Substring(serviceStartIndex, serviceEndIndex - serviceStartIndex);
+                    foreach (string subDomain in serviceSubDomains)
+                    {
+                        if (service == subDomain)
+                        {
+                            return uri.Host.Substring(0, accountEndIndex);
+                        }
+                    }
+                }
+            }
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// If path starts with a slash, remove it
+        /// </summary>
+        /// <param name="uri">The Uri.</param>
+        /// <returns>Sanitized Uri.</returns>
+        public static string GetSanitizedPath(this Uri uri) =>
+            (uri.AbsolutePath[0] == '/')
+            ? uri.AbsolutePath.Substring(1)
```

Comment:
Style nit - put the `?` and the `:` at the end of the line before and then indent the consequent and the alternate.

Reply:
That seems hard to read. I found this https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/651
How about starting the lines with ? and : and still indenting?

Reply:
`            (uri.AbsolutePath[0] == '/')
                ? uri.AbsolutePath.Substring(1)
                : uri.AbsolutePath;`

Reply:
That's an issue somebody filed in 2015 and it's still open because it's not the standard way folks write that in C#.   The following is preferred:

``` C#
(uri.AbsolutePath[0] == '/') ?
    uri.AbsolutePath.Substring(1) :
    uri.AbsolutePath;
```